src/rgw/rgw_main.cc

```diff
@@ -483,11 +481,10 @@ int radosgw_Main(int argc, const char **argv)
     store->set_luarocks_path(luarocks_path+"/"+g_conf()->name.to_str());
   }
 #ifdef WITH_RADOSGW_LUA_PACKAGES
-  rgw::sal::RadosStore *rados = dynamic_cast<rgw::sal::RadosStore*>(store);
-  if (rados) { /* Supported for only RadosStore */
+  if (store->get_name() == "rados") { /* Supported for only RadosStore */
```

Comment:
if we move the implementation of `install_packages` to the base class, we can remove the RADOS check:
```cpp
if (store->get_name() == "rados")
```

Reply:
Just so I'm clear, you want me to move the implementation (currently in RadosLuaManager in rgw_sal_rados.cc) to StoreLuaManager in rgw_sal_store.h (or in a new rgw_sal_store.cc) instead?

Reply:
rgw_sal.h  declares the pure virtual interface.
rgw_sal_store.{h,cc} defines default implementations that return -ENOENT.  (Or so I thought.)
rgw_sal_rados.cc defines the RADOS implementations.
rgw_sal_{dbstore,motr,...} have no implemenations at this time.

Reply:
ok. thanks.
so, the default implementation should go in the "sal_store"